src/TitleDetails/TitleDetailsContainer.js

```diff
@@ -64,16 +105,61 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
     [history, title.id, location.search],
   );
 
+  const onAddPiece = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(() => showCallout({
+          messageId: 'ui-receiving.piece.actions.addPiece.success',
+          type: 'success',
+          values: { title: title.title },
+        }))
+        .catch(() => showCallout({ messageId: 'ui-receiving.piece.actions.addPiece.error', type: 'error' }))
+        .finally(fetchPieces);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [mutator.orderPieces],
+  );
+
+  const onCheckIn = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(piece => {
+          showCallout({
+            messageId: 'ui-receiving.piece.actions.addPiece.success',
+            type: 'success',
+            values: { title: title.title },
+          });
+
+          return checkInItems([{
+            ...piece,
+            itemStatus: ITEM_STATUS.inProcess,
+          }], mutator.checkIn);
+        })
+        .catch(() => showCallout({ messageId: 'ui-orders.piece.actions.checkInItem.error', type: 'error' }))
+        .finally(fetchPieces);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [mutator.checkIn, mutator.orderPieces],
```

Comment:
I'm not sure if adding mutators as a callback dep is a good idea, sometimes it causes infinity loops because it could be a new object every render (we have no control on stripes-connect). So we use `// eslint-disable-next-line react-hooks/exhaustive-deps` to avoid linter to warning us regarding missing mutator deps.
However, if you adding them as a dep - you don't need to silent the linter.